pkg/auth/model/model.go

```diff
@@ -109,16 +126,165 @@ func (s *Statements) Scan(src interface{}) error {
+		Source:            u.Source,
+	}
+}
+
+//nolint
+func kvGroupPath(DisplayName string) string {
+	return kv.FormatPath(authPrefix, "groups", DisplayName)
+}
+
+//nolint
+func groupFromProto(pb *GroupData) *Group {
+	return &Group{
+		ID:          int(pb.Id),
+		CreatedAt:   pb.CreatedAt.AsTime(),
+		DisplayName: pb.DisplayName,
+	}
+}
+
+//nolint
+func protoFromGroup(g *Group) *GroupData {
+	return &GroupData{
+		Id:          int64(g.ID),
+		CreatedAt:   timestamppb.New(g.CreatedAt),
+		DisplayName: g.DisplayName,
+	}
+}
+
+//nolint
+func kvPolicyPath(DisplayName string) string {
+	return kv.FormatPath(authPrefix, "policies", DisplayName)
+}
+
+//nolint
+func policyFromProto(pb *PolicyData) *Policy {
+	return &Policy{
+		ID:          int(pb.Id),
+		CreatedAt:   pb.CreatedAt.AsTime(),
+		DisplayName: pb.DisplayName,
+		Statement:   *statementsFromProto(pb.Statements),
+	}
+}
+
+//nolint
+func protoFromPolicy(g *Policy) *PolicyData {
+	return &PolicyData{
+		Id:          int64(g.ID),
+		CreatedAt:   timestamppb.New(g.CreatedAt),
+		DisplayName: g.DisplayName,
+		Statements:  protoFromStatements(&g.Statement),
+	}
+}
+
+//nolint
+func kvCredentialPath(UserName string, AccessKeyID string) string {
+	return kv.FormatPath(authPrefix, UserName, "credentials", AccessKeyID)
+}
+
+//nolint
+func credentialFromProto(pb *CredentialData) *Credential {
+	return &Credential{
+		AccessKeyID:                   pb.AccessKeyId,
+		SecretAccessKey:               pb.SecretAccessKey,
+		SecretAccessKeyEncryptedBytes: pb.SecretAccessKeyEncryptedBytes,
+		IssuedDate:                    pb.IssuedDate.AsTime(),
+		UserID:                        pb.UserId,
+	}
+}
+
+//nolint
+func protoFromCredential(c *Credential) *CredentialData {
+	return &CredentialData{
+		AccessKeyId:                   c.AccessKeyID,
+		SecretAccessKey:               c.SecretAccessKey,
+		SecretAccessKeyEncryptedBytes: c.SecretAccessKeyEncryptedBytes,
+		IssuedDate:                    timestamppb.New(c.IssuedDate),
+		UserId:                        c.UserID,
+	}
+}
+
+//nolint
+func statementFromProto(pb *StatementData) *Statement {
+	return &Statement{
+		Effect:   pb.Effect,
+		Action:   pb.Action,
+		Resource: pb.Resource,
+	}
+}
+
+//nolint
+func protoFromStatement(s *Statement) *StatementData {
+	return &StatementData{
+		Effect:   s.Effect,
+		Action:   s.Action,
+		Resource: s.Resource,
+	}
+}
+
+//nolint
+func statementsFromProto(pb *StatementsData) *Statements {
+	statements := make(Statements, len(pb.Statement))
+	for i := range pb.Statement {
+		statements[i] = *statementFromProto(pb.Statement[i])
+	}
+	return &statements
+}
+
+//nolint
+func protoFromStatements(s *Statements) *StatementsData {
+	statements := make([]*StatementData, len(*s))
+	x := *s
+	for i := range *s {
+		statements[i] = protoFromStatement(&x[i])
+	}
+	return &StatementsData{
+		Statement: statements,
+	}
+}
+
+//nolint
+func kvUserToGroup(UserName string, DisplayName string) string {
```

Comment:
```suggestion
func kvUserToGroup(userName string, displayName string) string {
```

Reply:
Display name of the group? It's confusion to use just the property name without the type when there are multiple types with the same property.

Reply:
Here and elsewhere.

Reply:
done